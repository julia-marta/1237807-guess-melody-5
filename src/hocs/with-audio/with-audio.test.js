import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from "prop-types";
import withAudio from "./with-audio";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <React.Fragment>
      {children}
    </React.Fragment>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withAudio(MockComponent);

it(`should withAudio render correctly`, () => {
  const tree = renderer
    .create((
      <MockComponentWrapped isPlaying={true} src={``} />), {
      createNodeMock: () => {
        return {};
      }
    }
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
