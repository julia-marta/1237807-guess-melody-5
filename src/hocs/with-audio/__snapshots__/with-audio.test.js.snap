// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should withAudio render correctly 1`] = `<audio />`;
